adoc/headers/group.h

```diff
@@ -10,6 +10,8 @@ template <int Dimensions = 1> class group {
   static constexpr int dimensions = Dimensions;
   static constexpr memory_scope fence_scope = memory_scope::work_group;
 
+  group() = delete;
```

Comment:
I think this looks good, but should we do the same also for `sub_group`?

Reply:
Done in e41eda4b76077507f19e3d2cfcf700e877fd1f9d